lib/grplatt.gi

```diff
@@ -3335,57 +3335,254 @@ local c,n,deg,ind,core,i,j,sum;
 
 end);
 
+
+# utility function: Find subgroup of $G$, containing $G'$ of index up to
+# limit which is normal in the normalizer of G in P.
+BindGlobal("BoundedIndexAbelianized",function(P,G,limit)
+local d,ind,i,ma,ab,a,p,b,c,e,n;
+  d:=DerivedSubgroup(G);
+  ind:=IndexNC(G,d);
+  if ind=1 then return G;
+  # derived index small enough
+  elif ind<=limit then return d;
+  elif IsPrimeInt(ind) then return d;fi;
+
+  # make a p group
+  ind:=List(Collected(Factors(ind)),x->x[1]^x[2]);
+  Sort(ind);
+  if Length(ind)>1 then 
+    i:=0;
+    while i<Length(ind) and ind[i+1]<=limit do
+      i:=i+1;
+    od;
+    ind:=List(ind{Difference([1..Length(ind)],[i])},x->Factors(x)[1]);
+    for i in ind do
+      d:=ClosureSubgroup(d,SylowSubgroup(G,i));
+    od;
+    ind:=IndexNC(G,d);
+    if ind<=limit then return d;
+    elif IsPrimeInt(ind) then return d;fi;
+  fi;
+
+  # make elementary
+  p:=Factors(IndexNC(G,d))[1];
+  ma:=NaturalHomomorphismByNormalSubgroup(G,d);
+  a:=Image(ma,G);
+  ab:=AbelianInvariants(a);
+  if Length(Set(ab))>1 then
+    e:=LogInt(Exponent(a),p);
+    b:=fail;
```

Comment:
Observation: The code in this `if` statement (lines 3373-3394) is never executed by the Travis test suite.